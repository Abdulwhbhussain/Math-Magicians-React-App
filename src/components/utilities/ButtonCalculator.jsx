import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// styled-components
const ButtonStyled = styled.button`
  width: 25%;
  height: 100px;
  font-size: 1.5rem;
  font-weight: bold;
  background-color: ${(props) => {
    if (props.variant === 'orange') {
      return '#FF8C00';
    }
    return '#d3d3d3';
  }};
  color: '#000000';
  cursor: pointer;
  outline: none;
  &:hover {
    background-color: ${(props) => {
    if (props.variant === 'orange') {
      return '#FFA500';
    }
    return '#a9a9a9';
  }};
  
  }`;

// Calculator Button Reusable Styled Component
const ButtonCalculator = ({ name, variant }) => (
  <ButtonStyled variant={variant}>
    {name}
  </ButtonStyled>
);

ButtonCalculator.defaultProps = {
  variant: 'grey',
};

ButtonCalculator.propTypes = {
  name: PropTypes.string.isRequired,
  variant: PropTypes.string,
};

export default ButtonCalculator;
