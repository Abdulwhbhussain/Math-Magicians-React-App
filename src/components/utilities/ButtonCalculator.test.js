import React from 'react';
import renderer from 'react-test-renderer';
import ButtonCalculator from './ButtonCalculator';

it('renders correctly', () => {
  const tree = renderer
    .create(<ButtonCalculator className="four" name="4" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
