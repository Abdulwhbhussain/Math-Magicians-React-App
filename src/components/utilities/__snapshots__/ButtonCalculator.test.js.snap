// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="sc-aXZVg dDdpDQ"
  name="4"
>
  4
</button>
`;
